s3/replication/manager/src/s3replicationmanager/manager_routes.py

```diff
@@ -0,0 +1,225 @@
+                    cls=JobJsonEncoder)))
+        return web.json_response({'job': job.get_dict()}, status=201)
+
+
+@routes.get('/jobs/{job_id}')  # noqa: E302
+async def get_job(request):
+    """Get job attribute
+
+    Handler api to fetch job attributes
+
+    """
+    job_id = request.match_info['job_id']
+    job = request.app['all_jobs'].get_job(job_id)
+
+    if job is not None:
+        LOG.debug('Job found with job_id : {} '.format(job_id))
+        return web.json_response({"job": job.get_dict()}, status=200)
+    else:
+        LOG.debug('Job missing with job_id : {} '.format(job_id))
+        return web.json_response(
+            {'ErrorResponse': 'Job Not Found!'}, status=404)
+
+
+@routes.put('/jobs/{job_id}')  # noqa: E302
+async def update_job_attr(request):
+    """Update job attributes
+
+    Update attributes for job_id
+
+    """
+    job_record = await request.json()
+
+    # Get key
+    job_id = next(iter(job_record))
+    job = request.app['all_jobs'].get_job(job_id)
+
+    if job is None:
+        return web.json_response(
+            {'ErrorResponse': 'job is not present'}, status=404)
+    else:
+        request.app['all_jobs']._jobs[job.get_job_id()] = job_record
+        LOG.debug('Updated Job with job_id : {} '.format(job.get_job_id()))
+        LOG.debug(
+            'Update Job : {} '.format(
+                json.dumps(
+                    job,
+                    cls=JobJsonEncoder)))
+
+
+@routes.get('/jobs')  # noqa: E302
+async def list_jobs(request):
+    """List active jobs
+
+    Handler api to handle multiple query
+
+    """
+    url = request.path_qs
+    url_ob = urlparse(url)
+    query = parse_qs(url_ob.query)
+
+    progressing_jobs_obj = request.app['jobs_in_progress']
+    all_jobs_obj = request.app['all_jobs']
+
+    # LOG.debug(
+    #        'progressing jobs : {}'.format(list(progressing_jobs_obj.keys())))
+    # LOG.debug('all jobs : {}'.format(list(all_jobs_obj.keys())))
+
+    # Return in progress jobs
+    if 'inprogress' in query:
+        LOG.debug('InProgress query param: {}'.format(query['inprogress']))
+        return web.json_response(
+            {'inprogress jobs': json.dumps(
+                list(progressing_jobs_obj.get_keys()))})
+
+    # Return total job counts
+    elif 'count' in query:
+        return web.json_response(
+            {'count': progressing_jobs_obj.count() + all_jobs_obj.count()})
+
+    # Return requested jobs
+    elif 'prefetch' in query:
+        prefetch_count = int(query['prefetch'][0])
+        sub_id = query['subscriber_id'][0]
+        LOG.debug('sub_id is : {}'.format(sub_id))
+
+        # Validate subscriber and server request
+        if request.app['subscribers'].check_presence(sub_id):
+
+            # Remove prefetch_count entries from jobs and add to inprogress
+            if prefetch_count < all_jobs_obj.count():
+                add_inprogress = dict(
+                    list(
+                        all_jobs_obj.get_all_jobs())[
```

Comment:
How about collection having remove 'n' jobs and remove returning the removed job items like typical pop method

Reply:
Good suggestion. thanks. 
done.